tests/common/lease_test.go

```diff
@@ -132,12 +132,21 @@ func TestLeaseGrantAndList(t *testing.T) {
 						createdLeases = append(createdLeases, leaseResp.ID)
 					}
 
-					resp, err := cc.LeaseList()
-					require.NoError(t, err)
-					require.Len(t, resp.Leases, nc.leaseCount)
+					// Because we're not guarunteed to talk to the same member, wait for
+					// listing to eventually return true, either by the result propagaing
+					// or by hitting an up to date member.
+					leases := []clientv3.LeaseStatus{}
+					require.Eventually(t, func() bool {
+						resp, err := cc.LeaseList()
+						if err != nil {
+							return false
+						}
+						leases = resp.Leases
+						return len(resp.Leases) == nc.leaseCount
```

Comment:
Checking if number of leases matches is not correct here as we cannot distinguish intermediate vs final state. I would recommend to instead check Revision. 

We can save revision from last `Grant` call and we we wait until we get a read with the revision. Then we can verify if the response has expected number of leases.

Reply:
sounds good 👍

Reply:
apparently we do! - updated.